de.metas.contracts/src/test/java/de/metas/contracts/impl/ContractChangeBLTest.java

```diff
@@ -131,6 +133,31 @@ public void cancel_a_Contract_which_was_extended_using_a_date_from_extended_cont
 		assertSubscriptionProgress(extendedContract, 3);
 		assertThat(contract.getMasterEndDate()).isEqualTo(extendedContract.getMasterEndDate());
 	}
+	
+	@Test
+	public void closeContract_test()
+	{
+		final I_C_Flatrate_Term contract = prepareContractForTest(true);
+		Services.get(IFlatrateBL.class).extendContract(contract, true, true, null, null);
+		save(contract);
+
+		final I_C_Flatrate_Term extendedContract = contract.getC_FlatrateTerm_Next();
+		assertThat(extendedContract).isNotNull();
+		
+		final ContractChangeParameters contractChangeParameters = ContractChangeParameters.builder()
+				.changeDate(SystemTime.asDayTimestamp())
+				.isCloseInvoiceCandidate(true)
+				.isOnlyTerminateCurrentTerm(true)
+				.build();
+		
+		assertThatThrownBy(() -> {
+			contractChangeBL.cancelContract(contract, contractChangeParameters);
+		}).hasMessageContaining(ContractChangeBL.MSG_IS_NOT_ALLOWED_TO_TERMINATE_CURRENT_CONTRACT);
+		
```

Comment:
aaaah, now I figured you are testing two things.
please do two distinct tests for them..ofc you can extract the setup code into a common method..